Homework 10 - Chapter 13 - Nguyen Chinh Quan - 220066

In [83]:
import pandas as pd
import numpy as np

13.12.

In [84]:
sample_size_12 = 185 + 180 + 230 + 220 + 185
list_12 = [185, 180, 230, 220, 185]
print(sample_size_12)

1000


Flavors: Almond, vanilla, raspberry, caramel, hazelnut -> k = 5

In [85]:
EF_12 = 1000 / 5 

H0: p(almond) = p(vanilla) = p(raspberry) = p(caramel) = p(hazelnut) = 1/5


Ha: p(flavor) != 1/5 (for each flavor)

In [86]:
x_squared_12 = 0
for i in range(5):
    x_squared_12 += ((list_12[i] - EF_12)**2 / EF_12)
x_squared_12 

10.75

alpha = 0.05 -> x^2(0.05, 4) = 9.488
We see that 10.75 > 9.488, therefore there is evidence that customers who purchase flavor shots prefer one over the others.

13.20.

In [87]:
data_20 = pd.read_excel(r"D:\stat\Tex-_Datasets\Chapter 13\ex13-20soliris.xls")
data_20

Tier  Frequency
0     1        152
1     2        107
2     3         36
3     4         76

a.

- H0: p1 = 0.35, p2 = 0.28, p3 = 0.12, p4 = 0.25
- Ha: p(tier) != p(past records) (for each tier)

In [88]:
sample_size_20 = data_20.Frequency.sum()
sample_size_20
list_proportion_20 = np.array([0.35, 0.28, 0.12, 0.25])
list_20 = list_proportion_20*sample_size_20
x_squared_20 = 0
print(list_20)
for i in range(4):
    x_squared_20 += ((data_20.Frequency[i] - list_20[i])**2 / list_20[i])
print(x_squared_20)

[129.85 103.88  44.52  92.75]
8.52753176742395


- alpha = 0.05 -> x^2(0.05, 3) = 7.815
- We see that 8.528 > 7.815, therefore there is a shift in the proportion of users by tiers.

b. 
- Tier 1 is ultilized more.
- Tier 2 is ultilized more.
- Tier 3 is ultilized less.
- Tier 4 is ultilized less.

13.28.

In [89]:
data_28 = pd.read_excel(r"D:\stat\Tex-_Datasets\Chapter 13\ex13-28metro.xls")
data_28 

Category  2012 responses  2014 sample
0  Transportation             127          135
1          Growth             127          124
2           Crime             120          110
3         Economy              80           89
4      Government              67           62
5           Taxes              40           43
6          Social              33           39
7         Housing              39           45
8           Other              33           36

a.

In [90]:
sample_size_2012_28 = data_28["2012 responses"].sum()
sample_size_2014_28 = data_28["2014 sample"].sum()
print(sample_size_2012_28, sample_size_2014_28)

666 683


In [91]:
proportion_2012_28 = np.array(data_28["2012 responses"]) / sample_size_2012_28
proportion_2012_28

array([0.19069069, 0.19069069, 0.18018018, 0.12012012, 0.1006006 ,
       0.06006006, 0.04954955, 0.05855856, 0.04954955])

The proportion associated with each category for the 2012 responses: [0.19069069, 0.19069069, 0.18018018, 0.12012012, 0.1006006 , 0.06006006, 0.04954955, 0.05855856, 0.04954955]

b.

In [92]:
expected_2014_28 = proportion_2012_28 * sample_size_2014_28
expected_2014_28

array([130.24174174, 130.24174174, 123.06306306,  82.04204204,
        68.71021021,  41.02102102,  33.84234234,  39.9954955 ,
        33.84234234])

In [93]:
x_squared_28 = 0
for i in range(9):
    x_squared_28 += ((data_28["2014 sample"][i] - expected_2014_28[i])**2) / expected_2014_28[i]
x_squared_28

4.750296353258287

 - H0: p(transportation) = 0.19, p(growth) = 0.19, p(crime) = 0.18, p(economy) = 0.12, p(government) = 0.10, p(taxes) = 0.06, p(social) = 0.05, p(housing) = 0.06, p(other) = 0.05
 - Ha: p(category) != p(category of 2012) (for each category)
 - alpha = 0.05 -> x^2(0.05, 8) = 15.507
 - We see that 4.75 < 15.507, therefore there is no evidence that the true 2012 population proportions have changed in 2014.

12.42.

In [94]:
data_42 = pd.read_excel(r"D:\stat\Tex-_Datasets\Chapter 13\ex13-42casino.xls")
data_42

Casino       Game  Count
0        Bellagio  Blackjack     22
1        Bellagio      Poker     20
2        Bellagio   Roulette     38
3        Bellagio      Slots     66
4        Caesar's  Blackjack     30
5        Caesar's      Poker     38
6        Caesar's   Roulette     22
7        Caesar's      Slots     68
8   Golden Nugget  Blackjack     28
9   Golden Nugget      Poker     25
10  Golden Nugget   Roulette     21
11  Golden Nugget      Slots     81
12       Harrah's  Blackjack     38
13       Harrah's      Poker     25
14       Harrah's   Roulette     29
15       Harrah's      Slots     84

In [95]:
Casino = ["Bellagio", "Caesar's", "Golden Nugget", "Harrah's"]
Game = ["Blackjack", "Poker", "Roulette", "Slots"]
sample_size_42 = data_42["Count"].sum()
sample_size_42

635

In [96]:
x_squared_42 = 0
for i in Casino:
    for j in Game: 
        expected_42 = data_42[data_42["Game"] == j]["Count"].sum() * data_42[data_42["Casino"] == i]["Count"].sum() / sample_size_42
        x_squared_42 += ((data_42[(data_42["Game"] == j) & (data_42["Casino"] == i)].Count.item() - expected_42)**2) / expected_42
x_squared_42

18.63757818772916

- H0: The true proportion of gamblers at each game is the same for all casinos.
- Ha: The true proportion of gamblers at each game is not the same for all casinos.
- alpha = 0.05 -> x^2(0.05, 9) = 16.919
- We see that 18.638 > 16.919, therefore there is evidence to suggest that the true proportion of gamblers at each game is the same for all casinos.

13.46.

In [97]:
data_46 = pd.read_excel(r"D:\stat\Tex-_Datasets\Chapter 13\ex13-46homework.xls")
data_46

Girls or Boys Times Helping with Homework per Week  Count
0         Girls                       Less than once    259
1         Girls                         1 or 2 times    369
2         Girls                         3 or 4 times    254
3         Girls                      5 or more times    118
4          Boys                       Less than once    274
5          Boys                         1 or 2 times    335
6          Boys                         3 or 4 times    262
7          Boys                      5 or more times    129

In [98]:
Gender = ["Girls", "Boys"]
Time = ["Less than once", "1 or 2 times", "3 or 4 times", "5 or more times"]
sample_size_46 = data_46.Count.sum()
sample_size_46

2000

In [99]:
x_squared_46 = 0
for i in Gender:
    for j in Time:
        expected_46 = data_46[data_46["Girls or Boys"] == i]["Count"].sum() * data_46[data_46["Times Helping with Homework per Week"] == j]["Count"].sum() / sample_size_46
        x_squared_46 += ((data_46[(data_46["Girls or Boys"] == i) & (data_46["Times Helping with Homework per Week"] == j)]["Count"].item() - expected_46)**2) / expected_46
x_squared_46

2.6780938415804973

- H0: The number of times spent helping with homework is the same for boys and girls.
- Ha: The number of times spent helping with homework is different for boys and girls.
- alpha = 0.01 -> x^2(0.01, 3) = 11.345
- We see that 2.678 < 11.345, therefore there is no evidence that the number of times spent helping with homework is different for boys and girls.
- Using the chi-squared table, the bounds on the p-value associated with this test are (0.1, 0.9)

13.48.

In [100]:
data_48 = pd.read_excel(r"D:\stat\Tex-_Datasets\Chapter 13\ex13-48resort.xls")
data_48

Age Group       Resort Activity  Count
0               Younger than 16       Downhill skiing     24
1               Younger than 16  Cross-country skiing     35
2               Younger than 16          Snowboarding     23
3               Younger than 16           Snow tubing     48
4                         16-20       Downhill skiing     50
5                         16-20  Cross-country skiing     25
6                         16-20          Snowboarding     40
7                         16-20           Snow tubing     32
8                         20-30       Downhill skiing     21
9                         20-30  Cross-country skiing     27
10                        20-30          Snowboarding     27
11                        20-30           Snow tubing     32
12                        30-40       Downhill skiing     29
13                        30-40  Cross-country skiing     39
14                        30-40          Snowboarding     29
15                        30-40           Snow tubing     29
16  Greater than or equal to 40       Downhill skiing     26
17  Greater than or equal to 40  Cross-country skiing     37
18  Greater than or equal to 40          Snowboarding     35
19  Greater than or equal to 40           Snow tubing     30

In [121]:
Age = ['Younger than 16', '16-20', '20-30', '30-40', 'Greater than or equal to 40']
Activity = ["Downhill skiing", "Cross-country skiing", "Snowboarding", "Snow tubing"]
sample_size_48 = data_48.Count.sum()
sample_size_48

638

In [122]:
x_squared_48 = 0
for i in Age:
    for j in Activity:
        expected_48 = data_48[data_48["Age Group"] == i]["Count"].sum() * data_48[data_48["Resort Activity"] == j]["Count"].sum() / sample_size_48
        x_squared_48 += ((data_48[(data_48["Age Group"] == i) & (data_48["Resort Activity"] == j)]["Count"].item() - expected_48)**2) / expected_48
x_squared_48

27.30048869755236

- H0: Resort activity is independent on age group.
- Ha: Resort activity is independent on age group.
- alpha = 0.01 -> x^2(0.01, 12) = 26.217
- We see that 27.3 > 26.217, therefore there is evidence to suggest that resort activity is dependent on age group.